src/warnet/cln.py

```diff
@@ -96,29 +96,46 @@ def get_graph_channels(self) -> list[LNChannel]:
         short_channel_ids = {chan["short_channel_id"]: chan for chan in cln_channels}.keys()
         channels = []
         for short_channel_id in short_channel_ids:
-            node1, node2 = (
+            nodes = [
                 chans for chans in cln_channels if chans["short_channel_id"] == short_channel_id
-            )
-            channels.append(self.lnchannel_from_json(node1, node2))
+            ]
+            # CLN has only heard about one side of the channel
+            if len(nodes) == 1:
+                channels.append(self.lnchannel_from_json(nodes[0], None))
+                continue
+            channels.append(self.lnchannel_from_json(nodes[0], nodes[1]))
         return channels
 
     @staticmethod
     def lnchannel_from_json(node1: object, node2: object) -> LNChannel:
         assert node1["short_channel_id"] == node2["short_channel_id"]
         assert node1["direction"] != node2["direction"]
+        if not node1:
+            raise ValueError("node1 can't be None")
+
+        node2_policy = (
+            LNPolicy(
+                min_htlc=node2["htlc_minimum_msat"],
+                max_htlc=node2["htlc_maximum_msat"],
+                base_fee_msat=node2["base_fee_millisatoshi"],
+                fee_rate_milli_msat=node2["fee_per_millionth"],
+            )
+            if node2 is None
+            else None
+        )
+
         return LNChannel(
             node1_pub=node1["source"],
-            node2_pub=node2["source"],
+            node2_pub=node2["source"] if node2 else node1["destination"],
```

Comment:
could remove the `if` here since there will always be a `node1["destination"]` and we shouldnt need to worry that node2 might have a different source

Reply:
Obvious now you said it.